src/classes/ADDR_GoogleGeoAPI_Validator.cls

```diff
@@ -310,29 +130,368 @@ public with sharing class ADDR_GoogleGeoAPI_Validator implements ADDR_IValidator
     * @description Apex model of the location element of the Google JSON response.
     */
     public class Location {
         public Double lat;
         public Double lng;
     }
-    
+
     /*******************************************************************************************************
     * @description Apex model of the viewport element of the Google JSON response.
     */
     public class Viewport {
         public Location northeast;
         public Location southwest;
     }
-    
+
     /*******************************************************************************************************
     * @description Parses a JSON strings and returns a GoogleGeoResponse Apex object.
-    * @param String The JSON string.
+    * @param json The JSON string.
     * @return GoogleGeoResponse The Apex object the JSON string has been transformed into.
     */
     public static GoogleGeoResponse parse(String json) {
         return (GoogleGeoResponse) System.JSON.deserialize(json, GoogleGeoResponse.class);
     }
+
+    /**
+    * @description Address Verification Wrapper Class
+    */
+    public class AddressVerifier {
+
+        private Address__c address;
+        private String endPoint;
+        private Addr_Verification_Settings__c settings;
+        private GoogleGeoResponse googleResponse;
+        private HttpResponse rawResponse;
+
+        /** @description SubPremise from the original address to append to the Street Address (if not null) */
+        private String subPremise;
+
+        /** @description Constructor with the address to verify */
+        public AddressVerifier(Address__c addr) {
+            this.address = addr;
+        }
+
+        /**
+        * @description Pass an instance of the AddressVerificationSettings object
+        * @return Instance
+        * */
+        public AddressVerifier withAddressSettings(Addr_Verification_Settings__c settings) {
+            this.settings = settings;
+            return this;
+        }
+
+        /**
+        * @description Pass the endpoint to use
+        * @return Instance
+        */
+        public AddressVerifier withEndPoint(String endPoint) {
+            this.endPoint = endPoint;
+            return this;
+        }
+
+        /**
+       * @description Verify this Aaddress record and return an instance of this class
+       * @return Instance
+       */
+        public AddressVerifier verifyAddress() {
+
+            if (settings == null || settings.Auth_Token__c == null || endPoint == null) {
+                address.Verified__c = false;
+                address.API_Response__c = 'MISSING AUTHID OR ENDPOINT.  Please see the Address Verification menu in the NPSP Settings tab for more information';
+                return this;
+            }
+
+            if (!calloutToAddressService()) {
+                return this;
+            }
+
+            if (googleResponse.status != ADDR_Validator_UTIL.RESPONSE_OK || googleResponse.results == null) {
```

Comment:
I had left it in this method so that the calloutToAddressService method only handled the callout and casting the response. Any processing of a valid response (even "other" responses are not total failures) was handled by verifyAddress.